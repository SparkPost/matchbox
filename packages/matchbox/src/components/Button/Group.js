import React from 'react';
import PropTypes from 'prop-types';

const Group = ({ children, className = '' }) => (
  <div className={className}>
    {children}
  </div>
);

Group.propTypes = {
  children: PropTypes.node.isRequired
};

Group.displayName = 'Button.Group';

export default Group;
