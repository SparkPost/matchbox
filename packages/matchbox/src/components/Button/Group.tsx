import React from 'react';
import styled from 'styled-components';
import { Box } from '../Box';
import { group } from './styles';

export type ButtonGroupProps = {
  className?: string;
  children?: React.ReactNode;
};

const StyledGroup = styled(Box)`
  ${group}
`;

const Group = ({ children, className = '' }: ButtonGroupProps): JSX.Element => (
  <StyledGroup display="inline-flex" alignItems="center" className={className}>
    {children}
  </StyledGroup>
);

Group.displayName = 'Button.Group';

export default Group;
