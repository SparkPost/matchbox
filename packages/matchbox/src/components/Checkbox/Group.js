import React from 'react';
import PropTypes from 'prop-types';
import { Label } from '../Label';
import styles from './Checkbox.module.scss';

const Group = ({ children, label, required }) => (
  <div className={styles.Group}>
    {label && <Label className={styles.GroupLabel}>{label}{required && ' *'}</Label>}
    {children}
  </div>
);

Group.propTypes = {
  children: PropTypes.node.isRequired,
  label: PropTypes.node,
  required: PropTypes.bool
};

Group.displayName = 'Checkbox.Group';
export default Group;
