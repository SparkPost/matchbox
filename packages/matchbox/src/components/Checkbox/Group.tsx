import React from 'react';
import { Box } from '../Box';
import { Label } from '../Label';
import { Stack } from '../Stack';
import { OptionalLabel } from '../OptionalLabel';
import styled from 'styled-components';
import { margin, MarginProps } from 'styled-system';
import { pick } from '@styled-system/props';

const StyledGroup = styled('fieldset')`
  border: none;
  padding: 0;
  ${margin}
`;

type CheckboxGroupProps = {
  children: React.ReactNode;
  className?: string;
  'data-id'?: string;
  label?: React.ReactNode;
  required?: boolean;
  labelHidden?: boolean;
  optional?: boolean;
} & MarginProps;

function Group(props: CheckboxGroupProps): JSX.Element {
  const {
    'data-id': dataId,
    children,
    className,
    label,
    labelHidden,
    required,
    optional,
    ...rest
  } = props;
  const systemProps = pick(rest);

  return (
    <StyledGroup {...systemProps} className={className} data-id={dataId}>
      {label && (
        <Box width="100%">
          <Label as="legend" labelHidden={labelHidden}>
            <Box as="span" pr="200">
              {label}
            </Box>
            {required && (
              <Box as="span" pr="200" aria-hidden="true">
                *
              </Box>
            )}
            {optional && <OptionalLabel />}
          </Label>
        </Box>
      )}
      <Stack space="100">{children}</Stack>
    </StyledGroup>
  );
}

Group.displayName = 'Checkbox.Group';
export default Group;
