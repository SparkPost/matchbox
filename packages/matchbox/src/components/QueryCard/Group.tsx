import React from 'react';
import styled from 'styled-components';

const StyledWrapper = styled.div`
  & > div:not(:first-child) {
    margin-top: -1px;
    border-top-right-radius: 0;
    border-top-left-radius: 0;
  }

  & > div:first-child {
    border-bottom-right-radius: 0;
    border-bottom-left-radius: 0;
  }
`;

type QueryCardGroupProps = {
  children?: React.ReactNode;
};

const Group = React.forwardRef<HTMLDivElement, QueryCardGroupProps>(function QueryCard(
  props,
  userRef,
) {
  return <StyledWrapper ref={userRef}>{props.children}</StyledWrapper>;
});

Group.displayName = 'QueryCard.Group';
export default Group;
