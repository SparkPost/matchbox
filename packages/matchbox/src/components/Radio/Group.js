import React from 'react';
import PropTypes from 'prop-types';
import { Label } from '../Label';
import styles from './Radio.module.scss';

const Group = ({ children, label, required }) => (
  <div className={styles.Group}>
    {label && <Label className={styles.GroupLabel}>{label}{required && ' *'}</Label>}
    {children}
  </div>
);

Group.propTypes = {
  children: PropTypes.node.isRequired,
  label: PropTypes.node,
  required: PropTypes.bool
};

Group.displayName = 'Radio.Group';
export default Group;
