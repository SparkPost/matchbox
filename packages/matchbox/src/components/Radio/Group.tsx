import React from 'react';
import PropTypes from 'prop-types';
import { Box } from '../Box';
import { Label } from '../Label';
import { ScreenReaderOnly } from '../ScreenReaderOnly';
import { OptionalLabel } from '../OptionalLabel';
import { Stack } from '../Stack';
import { createPropTypes } from '@styled-system/prop-types';
import styled from 'styled-components';
import { margin, MarginProps } from 'styled-system';
import { pick } from '@styled-system/props';

const StyledGroup = styled('fieldset')`
  border: none;
  padding: 0;
  ${margin}
`;

export type RadioGroupProps = {
  children: React.ReactNode;
  className?: string;
  'data-id'?: string;
  label: React.ReactNode;
  labelHidden?: boolean;
  optional?: boolean;
  required?: boolean;
} & MarginProps;

function Group(props: RadioGroupProps): JSX.Element {
  const {
    'data-id': dataId,
    children,
    className,
    label,
    labelHidden,
    required,
    optional,
    ...rest
  } = props;
  const systemProps = pick(rest);

  return (
    <StyledGroup {...systemProps} aria-required={required} className={className} data-id={dataId}>
      {label && (
        <Box width="100%">
          <Label as="legend" labelHidden={labelHidden}>
            <Box as="span" pr="200">
              {label}
              {required && <ScreenReaderOnly>Required</ScreenReaderOnly>}
            </Box>
            {optional && <OptionalLabel />}
          </Label>
        </Box>
      )}
      <Stack space="100">{children}</Stack>
    </StyledGroup>
  );
}

Group.displayName = 'Radio.Group';
export default Group;
