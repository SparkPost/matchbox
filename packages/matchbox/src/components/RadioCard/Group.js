import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { margin } from 'styled-system';
import { createPropTypes } from '@styled-system/prop-types';
import { getChild } from '../../helpers/children';
import { Box } from '../Box';
import { Columns } from '../Columns';
import { Column } from '../Column';
import { Label } from '../Label';
import { OptionalLabel } from '../OptionalLabel';
import { Stack } from '../Stack';

import { pick } from '../../helpers/systemProps';

const Fieldset = styled.fieldset`
  border: none;
  padding: 0;
  ${margin}
`;

const breakpoints = ['xs', 'sm', 'md', 'lg', 'xl'];

const Group = React.forwardRef(function Group(props, userRef) {
  const {
    children,
    collapseBelow,
    disabled,
    id,
    label,
    labelHidden,
    orientation,
    optional,
    weight,
    ...rest
  } = props;

  const items = getChild('RadioCard', children, { disabled, weight });
  const systemProps = pick(rest, margin.propNames);

  return (
    <Fieldset data-id={rest['data-id']} id={id} ref={userRef} {...systemProps}>
      <Label as="legend" label={label} labelHidden={labelHidden}>
        {optional && <OptionalLabel />}
      </Label>

      {orientation === 'horizontal' && (
        <Columns space="300" collapseBelow={collapseBelow}>
          {items.map((item, i) => (
            <Column key={i}>{item}</Column>
          ))}
        </Columns>
      )}

      {orientation === 'vertical' && (
        <Stack space="300" collapseBelow={collapseBelow}>
          {items.map((item, i) => (
            <div key={i}>{item}</div>
          ))}
        </Stack>
      )}
    </Fieldset>
  );
});

Group.displayName = 'RadioCard.Group';

Group.propTypes = {
  collapseBelow: PropTypes.oneOf(breakpoints),
  'data-id': PropTypes.string,
  disabled: PropTypes.bool,
  label: PropTypes.string.isRequired,
  labelHidden: PropTypes.bool,
  optional: PropTypes.bool,
  orientation: PropTypes.oneOf(['horizontal', 'vertical']),
  weight: PropTypes.oneOf(['light', 'heavy']),
  ...createPropTypes(margin.propNames),
};

Group.defaultProps = {
  orientation: 'vertical',
  weight: 'light',
};

export default Group;
