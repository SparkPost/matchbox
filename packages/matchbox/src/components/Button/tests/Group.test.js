import React from 'react';
import Group from '../Group';
import { shallow } from 'enzyme';

describe('Group', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Group><button>Button 1</button><button>Button 2</button></Group>);
  });

  it('renders children correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('sets classname if passed', () => {
    wrapper.setProps({ className: 'group-of-buttons' });
    expect(wrapper).toMatchSnapshot();
  });
});
