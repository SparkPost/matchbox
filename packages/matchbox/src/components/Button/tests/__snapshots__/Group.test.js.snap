// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group renders children correctly 1`] = `
<div
  className="Group"
>
  <button>
    Button 1
  </button>
  <button>
    Button 2
  </button>
</div>
`;

exports[`Group sets classname if passed 1`] = `
<div
  className="Group group-of-buttons"
>
  <button>
    Button 1
  </button>
  <button>
    Button 2
  </button>
</div>
`;
